R/sensitivity-spider-plot.R

```diff
@@ -107,104 +146,240 @@ sensitivitySpiderPlot <- function(sensitivityCalculation,
+    titleSize      = 14L,
+    xAxisScale     = xAxisScale,
+    xLabel         = "Input parameter value [% of reference]",
+    yAxisScale     = yAxisScale,
+    yAxisTicks     = 10L,
+    yLabel         = NULL
+  )
+  plotConfiguration <- .updatePlotConfiguration(
+    defaultPlotConfiguration, spiderPlotConfiguration
+  )
+
   # getting the scales right
   data <- dplyr::mutate(data,
     ParameterFactor = ParameterFactor * 100,
     PercentChangePK = PercentChangePK + 100
   )
 
-  plot <- ggplot(
-    data,
-    aes(x = ParameterFactor, y = PercentChangePK, group = ParameterPath)
-  ) +
-    geom_line(
-      aes(group = ParameterPath, color = as.factor(ParameterPath)),
-      size = 1.4,
-      alpha = 0.7,
-      na.rm = TRUE
-    ) +
-    geom_point(size = 2, shape = 21, na.rm = TRUE)
+  if (yAxisType == "percent") {
+    yColumn <- sym("PercentChangePK")
+    data$Unit <- "% of reference"
+  } else {
+    yColumn <- sym("PKParameterValue")
+  }
 
-  if (xAxisLog) {
-    plot <- plot + scale_x_log10()
+  # filter according to value limits
+  if (!is.null(plotConfiguration$yValuesLimits)) {
+    data <- dplyr::filter(data, !!yColumn >= plotConfiguration$yValuesLimits[1])
+    data <- dplyr::filter(data, !!yColumn <= plotConfiguration$yValuesLimits[2])
+  }
+  if (!is.null(plotConfiguration$xValuesLimits)) {
+    data <- dplyr::filter(data, ParameterFactor >= plotConfiguration$xValuesLimits[1])
+    data <- dplyr::filter(data, ParameterFactor <= plotConfiguration$xValuesLimits[2])
   }
 
-  if (yAxisLog) {
-    plot <- plot + scale_y_log10()
-  } else {
-    # how many labels on the Y-axis?
-    nBreaks <- labeling::extended(
-      min(na.omit(data$PercentChangePK)),
-      max(na.omit(data$PercentChangePK)),
-      m = 10L
-    )
+  # create plots for each PK parameter
+  plotList <- purrr::map(
+    unique(data$PKParameter),
+    ~ {
+      dataSubset <- dplyr::filter(data, PKParameter == .x)
+      baseDataSubset <- dplyr::filter(dataSubset, ParameterFactor == 100)
+
+      if (isTRUE(yAxisFacetScales == "fixed")) {
+        if (is.null(plotConfiguration$yLabel)) {
+          plotConfiguration$yLabel <- switch(yAxisType,
+            "percent"  = "PK-Parameter value [% of reference]",
+            "absolute" = "PK-Parameter value",
+            "PK-Parameter value"
+          )
+        }
+        pBreaks <- .calculateBreaks(data[, yColumn],
+          m = plotConfiguration$yAxisTicks,
+          Q = c(0, 100, 200, 300)
+        )
+        pLimits <- .calculateLimits(data[, yColumn])
+      } else {
+        plotConfiguration$yLabel <- paste0(
+          dataSubset$PKParameter[1], " [", dataSubset$Unit[1], "]"
+        )
+        pBreaks <- .calculateBreaks(
+          values = dataSubset[, yColumn],
+          m = plotConfiguration$yAxisTicks
+        )
+        pLimits <- NULL
+      }
+
+      if (!is.null(plotConfiguration$yAxisLimits)) {
+        pLimits <- plotConfiguration$yAxisLimits
+      }
+
+      plot <- ggplot(
```

Comment:
Please add comments (where it makes sense) what is being done in the next step. It will make debugging and adjusting easier.

Reply:
@rengelke Last to-do ;)

Reply:
Yes! :) Now the comments are quite detailed.